src/dodal/log.py

```diff
@@ -3,32 +3,59 @@
 import logging
 from os import environ
 from pathlib import Path
-from typing import List, Optional
+from typing import List, Optional, Tuple, Union
 
 from bluesky.log import config_bluesky_logging
 from bluesky.log import logger as bluesky_logger
+from graypy import GELFTCPHandler
 from ophyd.log import config_ophyd_logging
 from ophyd.log import logger as ophyd_logger
 
 LOGGER = logging.getLogger("Dodal")
-LOGGER.setLevel(logging.DEBUG)  # default logger to log everything
+LOGGER.setLevel(logging.DEBUG)
 ophyd_logger.parent = LOGGER
 bluesky_logger.parent = LOGGER
 
 
+class BeamlineFilter(logging.Filter):
+    beamline: Union[str, None] = environ.get("BEAMLINE")
+
+    def filter(self, record):
+        record.beamline = self.beamline if self.beamline else "dev"
+        return True
+
+
+beamline_filter = BeamlineFilter()
+
+
+def set_beamline(beamline_name: str):
+    """Set the beamline on all subsequent log messages."""
+    beamline_filter.beamline = beamline_name
+
+
 def set_up_logging_handlers(
-    logging_level: Optional[str] = "INFO", dev_mode: bool = False
+    logging_level: Optional[str] = "INFO",
+    dev_mode: bool = False,
+    logging_path: Optional[Path] = None,
 ) -> List[logging.Handler]:
     """Set up the logging level and instances for user chosen level of logging.
-
-    Mode defaults to production and can be switched to dev with the --dev flag on run.
+    Args:
+        logging_level: The level of logs that should be saved to file/graylog. Defaults to INFO.
+        dev_mode: True if in dev mode, will not log to graylog in dev. Defaults to False.
+        logging_path: The location to store log files, if left as None then puts them in the default location.
```

Comment:
Would it be reasonable to enable one or the other out of file logging and graylog?
It feels more elegant to have a series of optional routines for setting up various logging handlers and call each of them (or not) according to config/args

Reply:
I can't see a use case for only using one at the moment and I like the idea of a default simple function that gives you broad logging into a standard place. I'm happy to split it up a bit though so if people want to customise it they can call individual functions.

Reply:
Sounds good, a use case I had in mind was running inside a container. Logging to a file requires setting up a mount to somewhere persistent, which is enough effort that you may as well just use graylog.